### PR TITLE
Weets api integration

### PR DESCRIPTION
**Pull request by srwilliamg**
## Summary

- The configurations for axios interceptors was added
- A new helper for asynchronous calls was added
- A new endpoint "witter" was added, this endpoint provides a new geek joke every time is called: GET localhost:8081

## Trello Card
https://trello.com/c/2U17WpTb/32-servicios-promises-witter

### CODE REVIEW

File: app/services/witter.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+const axios = require('axios');
+
+const GEEK_JOKES_URL = 'https://geek-jokes.sameerkumar.website/api?format=json';
```
Comment (blueJayWays) [inline app/services/witter.js]: Use env vars to set external services endpoints. In `app/index` you will find the config.

File: config/axios.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+const axiosConfig = module.exports;
+
+axiosConfig.initAxios = () => {
+  const getResponse = response => {
+    const { data: responseData } = response;
+    return responseData;
+  };
+
+  const rejectResponse = error => {
```
Comment (blueJayWays) [inline config/axios.js:13]: you could log the errors. Check this file: `app/logger`

File: app/services/witter.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+const axios = require('axios');
```
Comment (blueJayWays) [inline app/services/witter.js:1]: If you're creating an axios file in `config/axios.js` then you don't need to require axios here.

File: config/axios.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    const { message, response: { status } = { status: 500 } } = error;
```
Comment (blueJayWays) [inline config/axios.js:14]: there's a file `app/errors.js` that can help you with error messages. You could use it.

**Review by blueJayWays — CHANGES_REQUESTED** [review]

File: app/services/witter.js (hunk 1 above)
Comment (srwilliamg) [reply app/services/witter.js]: Solved

File: config/axios.js (hunk 2 above)
Comment (srwilliamg) [reply config/axios.js:13]: solved

File: app/services/witter.js (hunk 3 above)
Comment (srwilliamg) [reply app/services/witter.js:1]: config/axios.js does not export axios

File: config/axios.js (hunk 4 above)
Comment (srwilliamg) [reply config/axios.js:14]: done

File: config/index.js [hunk 5]
```diff
@@ -44,6 +44,9 @@ const config = {
       apiDate: process.env.API_DATE || 'X-API-Date',
       packageVersion: process.env.PACKAGE_VERSION || 'X-Package-Version',
       nodeVersion: process.env.NODE_VERSION || 'X-Node-Version'
+    },
+    external_services: {
+      geek_jokes_url: 'https://geek-jokes.sameerkumar.website/api?format=json'
```
Comment (blueJayWays) [inline config/index.js]: the idea is not to hardcode the URL
you should create a env var named `GEEK_JOKES_URL` and get the value using [this library](https://www.npmjs.com/package/dotenv) that is already installed in the project.

**Review by blueJayWays — CHANGES_REQUESTED** [review]

File: config/index.js (hunk 5 above)
Comment (srwilliamg) [reply config/index.js]: Solved :)

**Comment by blueJayWays** [comment]
Looks good to me  👍

**Pull request merged**